components/persistor/test/persist_banking_s3.js

```diff
@@ -0,0 +1,205 @@
+/*
+ * Banking example shows PersistObjectTemplate
+ * with remote object storage using S3RemoteDocClient
+ */
+
+const sinon = require('sinon');
+const S3RemoteDocClient = require('../dist/lib/remote-doc/remote-doc-clients/S3RemoteDocClient').S3RemoteDocClient;
+const expect = require('chai').expect;
+const AssertionError = require('chai').AssertionError;
+const ObjectTemplate = require('@havenlife/supertype').default;
+const PersistObjectTemplate = require('../dist/index.js')(ObjectTemplate, null, ObjectTemplate);
+const logLevel = process.env.logLevel || 'debug';
+
+PersistObjectTemplate.debugInfo = 'api;conflict;write;read;data';//'api;io';
+PersistObjectTemplate.debugInfo = 'conflict;data';//'api;io';
+PersistObjectTemplate.logger.setLevel(logLevel);
+
+const Customer = PersistObjectTemplate.create('Customer', {
+    init: function (bankingDocumentValue) {
+        this.bankingDocument = bankingDocumentValue;
+    },
+    bankingDocument: {
+        type: String,
+        isRemoteObject: true,
+        remoteKeyBase: 'test-remote-key',
+        value: null
+    }
+});
+
+const schema = {
+    Customer: {
+        documentOf: 'pg/customer'
+    }
+}
+
+function clearCollection(template) {
+    let collectionName = template.__collection__.match(/\//) ? template.__collection__ : 'mongo/' + template.__collection__;
+    if (collectionName.match(/pg\/(.*)/)) {
+        collectionName = RegExp.$1;
+        return PersistObjectTemplate.dropKnexTable(template)
+            .then(function () {
+                return PersistObjectTemplate.synchronizeKnexTableFromTemplate(template).then(function() {return 0});
+            });
+    } else
+        throw 'Invalid collection name ' + collectionName;
+}
+
+describe('Banking from pgsql Example persist_banking_s3', function () {
+    let knex;
+
+    afterEach(function() {
+        sinon.restore();
+    });
+
+    let noBankingDocumentCustomer;
+
+    before(async () => {
+        knex = require('knex')({
+            client: 'pg',
+            connection: {
+                host: process.env.dbPath,
+                database: process.env.dbName,
+                user: process.env.dbUser,
+                password: process.env.dbPassword,
+
+            }
+        });
+        PersistObjectTemplate.setRemoteDocConnection({
+            persistorBucketName: 'test-bucket-persistor',
+            persistorRemoteDocEnvironment: 'S3'
+        });
+        PersistObjectTemplate.setDB(knex, PersistObjectTemplate.DB_Knex,  'pg');
+        PersistObjectTemplate.setSchema(schema);
+        PersistObjectTemplate.performInjections(); // Normally done by getTemplates
+
+        this.timeout(4000);
+
+        const schemaTable = 'index_schema_history';
+        await knex.schema.dropTableIfExists(schemaTable)
+        const count = await clearCollection(Customer);
+        expect(count).to.equal(0);
+
+        noBankingDocumentCustomer = new Customer();
+
+        PersistObjectTemplate.begin();
```

Comment:
Looks lie, begin and end are deprecated.

```PERSISTOR_LEGACY: begin: Use beginTransaction for alternative approaches```

Reply:
Yeah.. I basically copied the other `persist_banking_*` test and modified it for this. I didn't want to take the extra time to rewrite everything as I imagine its part of the teams tech debt to work on that.